source/renderer/app/components/navigation/Navigation.js

```diff
@@ -48,45 +50,50 @@ export default class Navigation extends Component<Props> {
       activeItem,
       items,
     } = this.props;
+
     return (
       <div className={styles.component}>
-        {items.map(({ id, icon, label, isLegacy, hasNotification, ...item }) =>
-          item.type === 'dropdown'
-            ? (!isLegacy && (
-                <NavDropdown
-                  key={id}
-                  label={label}
-                  icon={icon}
-                  isActive={isActiveNavItem(id, item)}
-                  onChange={i => onNavItemClick(i)}
-                  isLegacy={isLegacy}
-                  activeItem={activeItem}
-                  options={item.options}
-                  hasNotification={hasNotification}
-                />
-              )) ||
-              (isLegacy && (
-                <NavButton
-                  key={id}
-                  className={id}
-                  label={label}
-                  icon={icon}
-                  isActive={isActiveNavItem(id, item)}
-                  onClick={() => onNavItemClick(id)}
-                  hasNotification={hasNotification}
-                />
-              ))
-            : !isLegacy && (
-                <NavButton
-                  key={id}
-                  className={id}
-                  label={label}
-                  icon={icon}
-                  isActive={isActiveNavItem(id, item)}
-                  onClick={() => onNavItemClick(id)}
-                  hasNotification={hasNotification}
-                />
-              )
+        {items.map(
+          ({ id, icon, label, isLegacy, hasNotification, ...item }) => {
+            if (includes(ITN_LEGACY_WALLET_EXCLUDED_NAV_ITEMS, id) && isLegacy)
```

Comment:
This component is used by wallets and staking. Wouldn't it be better to have this logic, which applies only for wallets, inside [WalletNavigation.js](https://github.com/input-output-hk/daedalus/blob/e395bf17fcf7f2996d8292396b976dcaf381316e/source/renderer/app/components/wallet/navigation/WalletNavigation.js#L83-L126)?

Reply:
@daniloprates yes that would be better. I will improve that